File: src/test/java/com/tabnine/testUtils/HealthCheckTestUtils.java
package com.tabnine.testUtils;

import com.intellij.openapi.application.ApplicationManager;
import com.tabnine.binary.requests.config.CloudConnectionHealthStatus;
import com.tabnine.binary.requests.config.StateResponse;
import com.tabnine.lifecycle.BinaryStateChangeNotifier;

public class HealthCheckTestUtils {

  public static void notifyHealthStatusLoop(
      CloudConnectionHealthStatus cloudConnectionHealthStatus, int times) {
    for (int i = 0; i < times; i++) {
      notifyHealthStatus(cloudConnectionHealthStatus);
    }
  }

  public static void notifyHealthStatus(CloudConnectionHealthStatus cloudConnectionHealthStatus) {
    ApplicationManager.getApplication()
        .getMessageBus()
        .syncPublisher(BinaryStateChangeNotifier.STATE_CHANGED_TOPIC)
        .stateChanged(new StateResponse(null, null, null, cloudConnectionHealthStatus));
  }
}

File: Common/src/test/java/com/tabnine/testUtils/HealthCheckTestUtils.kt
package com.tabnineCommon.testUtils

import com.intellij.openapi.application.ApplicationManager
import com.tabnineCommon.binary.requests.config.CloudConnectionHealthStatus
import com.tabnineCommon.binary.requests.config.StateResponse
import com.tabnineCommon.lifecycle.BinaryStateChangeNotifier

object HealthCheckTestUtils {
    @JvmStatic
    fun notifyHealthStatus(
        cloudConnectionHealthStatus: CloudConnectionHealthStatus?,
        times: Int
    ) {
        for (i in 0 until times) {
            notifyHealthStatus(cloudConnectionHealthStatus)
        }
    }

    @JvmStatic
    fun notifyHealthStatus(cloudConnectionHealthStatus: CloudConnectionHealthStatus?) {
        ApplicationManager.getApplication()
            .messageBus
            .syncPublisher(BinaryStateChangeNotifier.STATE_CHANGED_TOPIC)
            .stateChanged(StateResponse(null, null, null, cloudConnectionHealthStatus!!))
    }
}
